homeassistant/components/sigfox/sensor.py

```diff
@@ -126,7 +126,11 @@ def get_last_message(self):
         url = urljoin(API_URL, device_url)
         response = requests.get(url, auth=self._auth, timeout=10)
         data = json.loads(response.text)["data"][0]
-        payload = bytes.fromhex(data["data"]).decode("utf-8")
+        try:
+            payload = bytes.fromhex(data['data']).decode('utf-8')
```

Comment:
This integration should have its api interaction and data parsing extracted to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#5-communication-with-devicesservices

Reply:
e.g. https://github.com/hecko/pySigfox